tests/Umbraco.Tests.Integration/Umbraco.Infrastructure/Services/FileServiceTests.cs

```diff
@@ -14,91 +15,196 @@ namespace Umbraco.Cms.Tests.Integration.Umbraco.Infrastructure.Services;
 [UmbracoTest(Database = UmbracoTestOptions.Database.NewSchemaPerTest)]
 internal sealed class FileServiceTests : UmbracoIntegrationTest
 {
-    private IFileService FileService => GetRequiredService<IFileService>();
+    private ITemplateService TemplateService => GetRequiredService<ITemplateService>();
+    private IUserService UserService => GetRequiredService<IUserService>();
+
+    private Guid _userKey;
 
     [SetUp]
     public void SetUp()
     {
-        var fileSystems = GetRequiredService<FileSystems>();
-        var viewFileSystem = fileSystems.MvcViewsFileSystem!;
-        foreach (var file in viewFileSystem.GetFiles(string.Empty).ToArray())
-        {
-            viewFileSystem.DeleteFile(file);
-        }
+        _userKey = UserService.GetUserById(-1).Key;
     }
 
     [Test]
     public void Create_Template_Then_Assign_Child()
     {
-        var child = FileService.CreateTemplateWithIdentity("Child", "child", "test");
-        var parent = FileService.CreateTemplateWithIdentity("Parent", "parent", "test");
+        var attempt = TemplateService.CreateAsync("Child", "child", "test", _userKey).Result;
```

Comment:
What was the reason for the update of these tests?  It's not clear to me how they relate to the main changes in the PR.